### DISCUSSION TITLE
Cannot run artisan export from route call, i.e. Artisan::call('export')

### DISCUSSION THREAD

**Discussion by refocusTheory** [spatie/laravel-export#94] [Q&A]
When called from a route closure, it says  command not exist, am I doing something wrong?....
          Route::get('/export-test', function() {
            Artisan::call('export');
            return "Exported!";
        });

**Comment by vlastimilvasek** [comment]
I was unlucky with following too:
Route::get('/export', function (Exporter $exporter) {
    $exporter->export();   
});

error: File or directory [public] not found

**Comment by vlastimilvasek** [comment]
Strange things happen. I nearly reproduce export command and make my own (site:export). Artisan::call('site:export') works but same error : File or directory [public] not found

namespace App\Console\Commands;

use Illuminate\Console\Command;
use Spatie\Export\Exporter;

class ExportSite extends Command
{
    /**
     * The name and signature of the console command.
     *
     * @var string
     */
    protected $signature = 'site:export';

    /**
     * The console command description.
     *
     * @var string
     */
    protected $description = 'Vygeneruje statický web.';

    /**
     * Execute the console command.
     */
    public function handle(Exporter $exporter)
    {
        $this->info('Exporting site...');

        $exporter->export();

        if (config('export.disk')) {
            $this->info('Files were saved to disk `'.config('export.disk').'`');
        } else {
            $this->info('Files were saved to `dist`');
        }
    }
}
